contributors/devel/conformance-tests.md

```diff
@@ -1,110 +1,150 @@
-(i.e. sharing writable host /tmp with a container)
-- should be stable and run consistently
-- cannot skip providers (there should be no Skip like directives added to the test), 
-especially in the Nucleus or Application layers as described 
-[here](https://github.com/kubernetes/community/blob/master/contributors/devel/architectural-roadmap.md).
-- cannot test cloud provider specific features (i.e. GCE monitoring, S3 Bucketing, ...)
-- should work with default settings for all configuration parameters 
-(example: the default list of admission plugins should not have to be tweaked for passing conformance).
-- cannot rely on any binaries that are not required for the
-linux kernel or for a kubelet to run (i.e. git)
-- any container images used in the test must support all architectures for which kubernetes releases are built
-
-### Conformance Test Version Skew Policy
+The Kubernetes Conformance test suite is a subset of e2e tests that SIG
+Architecture has approved to define the core set of interoperable features that
+all conformant Kubernetes clusters must support. The tests verify that the
+expected behavior works as a user might encounter it in the wild.
+
+The process to add new conformance tests is intended to decouple the development
+of useful tests from their promotion to conformance:
+- Contributors write and submit e2e tests, to be approved by owning SIGs
+  - Tests should begin by covering the most important and visible aspects of the
+    function
+- Tests are proven to meet the [conformance test requirements] by review
+  and by accumulation of data on flakiness and reliability
+- A follow up PR is submitted to [promote the test to conformance](#promoting-tests-to-conformance)
+
+NB: This should be viewed as a living document in a few key areas:
+- The desired set of conformant behaviors is not adequately expressed by the
+  current set of e2e tests, as such this document is currently intended to
+  guide us in the addition of new e2e tests than can fill this gap
+- This document currently focuses solely on the requirements for GA,
+  non-optional features or APIs. The list of requirements will be refined over
+  time to the point where it as concrete and complete as possible.
+- There are currently conformance tests that violate some of the requirements
+  (ie: require privileged access), we will be categorizing these tests and
+  deciding what to do once we have a better understanding of the situation
+- Once we resolve the above issues, we plan on identifying the appropriate areas
+  to relax requirements to allow for the concept of conformance Profiles that
+  cover optional or additional behaviors
+
+## Conformance Test Requirements
+
+Conformance tests currently test only GA, non-optional features or APIs. More
+specifically, a test is eligible for promotion to conformance if:
+
+- it tests only GA, non-optional features or APIs (ie: no alpha or beta
```

Comment:
Do we want to say anything about
- addons?
- admission plugins?

Reply:
never mind, you said admission plugins below (line 57)

Reply:
We need to qualify that the test may create or set objects or fields that are alpha or beta that bypass policies or protections that are not yet GA, but which may reasonably be enabled on a conformant cluster.  Examples are attempting to set pod security policy to allow certain actions in test setup (and warning or allowing failure), or setting not-yet-GA scheduler annotations that may be used by admission control to protect against certain scheduling features.

Obviously we don’t need to do all of these up front, but I want conformant implementations to have flexibility to request certain security limiting choices to be relaxed by the test.

 if the test may be reasonably expected

Reply:
@smarterclayton I want to understand what you're suggesting. I my view the tests will define, for the most part, what an end-user will see if they try to perform certain actions and a conformant impl must adhere to that definition. How it achieves that goal is really up to it since our testing isn't (and probably can't) get under the covers to know what's going on with each possible config knob. Having said that, if the tests explicitly talk about certain features or config settings then those must match what the impl has and does. An impl can not be consider conformant if the tests says "x must be set to y" if the impl either rejects that request or changes x to something other than y.  So, if (outside the scope of what the tests define) an impl uses alpha features that's its own choice that we have no say over.  Is this consistent with that you're saying?

Reply:
@smarterclayton 

I think we need to move towards the tests being run in a blackbox fashion, where cluster configuration and maybe even namespace creation and configuration are performed outside the context of the tests. Otherwise, depending on how authorization, security, networking, and other policies have been configured, various tests may fail.

If at all possible, I'd also like to define a set of functionality that really should work out of the box in all reasonable production-worthy environments (so, maybe not minikube or kubernetes-in-docker), including ones that are fairly locked down, which is the reason for the "non-privileged" point below.

As these guidelines are written to be aspirational and intended to trigger discussion about exceptions, I'd prefer to not add these types of caveats to this doc at this time. Let's assess the whole body of tests for divergence from these principles and then re-evaluate.

cc @spiffxp @dims

Reply:
@bgrant0607 we need a list of principles in a living document. which is exactly what this doc is. So i am quite ok with listing them here since we have the caveat that this is not the full list and may be modified in the future. we have to start somewhere!

Reply:
I'm going to put this under "example of areas we _may_ want to relax" so it's not just living in a PR